sdk/tests/core/test_az_mqtt5_rpc_client.c

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "az_span_private.h"
+#include "az_test_definitions.h"
+#include <azure/core/az_mqtt5_rpc_client.h>
+#include <azure/core/az_result.h>
+#include <azure/core/internal/az_span_internal.h>
+
+#include <setjmp.h>
+#include <stdarg.h>
+#include <stddef.h>
+
+#include <cmocka.h>
+
+#define TEST_COMMAND_NAME "test_command_name"
+#define TEST_MODEL_ID "test_model_id"
+#define TEST_CLIENT_ID "test_client_id"
+#define TEST_SERVER_ID "test_server_id"
+#define TEST_RESPONSE_TOPIC                                                   \
+  "vehicles/" TEST_MODEL_ID "/commands/" TEST_SERVER_ID "/" TEST_COMMAND_NAME \
+  "/__for_" TEST_CLIENT_ID "\0"
+#define TEST_REQUEST_TOPIC \
+  "vehicles/" TEST_MODEL_ID "/commands/" TEST_SERVER_ID "/" TEST_COMMAND_NAME "\0"
+#define TEST_SUBSCRIPTION_TOPIC \
+  "vehicles/" TEST_MODEL_ID "/commands/+/" TEST_COMMAND_NAME "/__for_" TEST_CLIENT_ID "\0"
+
+static az_mqtt5_rpc_client test_rpc_client;
+
+static void test_az_mqtt5_rpc_client_options_default_success(void** state)
+{
+  (void)state;
+
+  az_mqtt5_rpc_client_options options = az_mqtt5_rpc_client_options_default();
+
+  assert_int_equal(options.subscribe_timeout_in_seconds, AZ_MQTT5_RPC_DEFAULT_TIMEOUT_SECONDS);
+  assert_int_equal(options.publish_timeout_in_seconds, AZ_MQTT5_RPC_DEFAULT_TIMEOUT_SECONDS);
+}
+
+static void test_az_rpc_client_init_no_options_success(void** state)
+{
+  (void)state;
+
+  char subscription_topic_buffer[256];
+  char response_topic_buffer[256];
+  char request_topic_buffer[256];
+
+  assert_int_equal(
+      az_rpc_client_init(
+          &test_rpc_client,
+          AZ_SPAN_FROM_STR(TEST_CLIENT_ID),
+          AZ_SPAN_FROM_STR(TEST_MODEL_ID),
+          AZ_SPAN_FROM_STR(TEST_COMMAND_NAME),
+          AZ_SPAN_FROM_BUFFER(response_topic_buffer),
+          AZ_SPAN_FROM_BUFFER(request_topic_buffer),
+          AZ_SPAN_FROM_BUFFER(subscription_topic_buffer),
+          NULL),
+      AZ_OK);
+
+  assert_true(az_span_is_content_equal(
+      test_rpc_client._internal.client_id, AZ_SPAN_FROM_STR(TEST_CLIENT_ID)));
+  assert_true(az_span_is_content_equal(
+      test_rpc_client._internal.model_id, AZ_SPAN_FROM_STR(TEST_MODEL_ID)));
+  assert_true(az_span_is_content_equal(
+      test_rpc_client._internal.command_name, AZ_SPAN_FROM_STR(TEST_COMMAND_NAME)));
+}
+
+static void test_az_rpc_client_init_options_success(void** state)
+{
+  (void)state;
+
+  char subscription_topic_buffer[256];
+  char response_topic_buffer[256];
+  char request_topic_buffer[256];
+  az_mqtt5_rpc_client_options options = az_mqtt5_rpc_client_options_default();
+  options.subscribe_timeout_in_seconds = 5;
+  options.publish_timeout_in_seconds = 3;
+
+  assert_int_equal(
+      az_rpc_client_init(
```

Comment:
test for client ID, command name, etc as NULL/EMPTY?

Reply:
Will add that with the custom topics PR since that will change what the behavior should be